Sources/Plasma/CoreLib/hsBitVector.h

```diff
@@ -116,137 +116,114 @@ class hsBitVector {
 
 inline hsBitVector::hsBitVector(const hsBitVector& other)
 {
-    if( 0 != (fNumBitVectors = other.fNumBitVectors) )
-    {       
+    if ((fNumBitVectors = other.fNumBitVectors)) {
         fBitVectors = new uint32_t[fNumBitVectors];
-        int i;
-        for( i = 0; i < fNumBitVectors; i++ )
+        for (uint32_t i = 0; i < fNumBitVectors; i++)
             fBitVectors[i] = other.fBitVectors[i];
+    } else {
+        fBitVectors = nullptr;
     }
-    else
-        fBitVectors = nil;
 }
 
 inline bool hsBitVector::Empty() const
 {
-    int i;
-    for( i = 0; i < fNumBitVectors; i++ )
-    {
-        if( fBitVectors[i] )
+    for (uint32_t i = 0; i < fNumBitVectors; i++ ) {
+        if (fBitVectors[i])
             return false;
     }
     return true;
 }
 
 inline bool hsBitVector::Overlap(const hsBitVector& other) const
 {
-    if( fNumBitVectors > other.fNumBitVectors )
+    if (fNumBitVectors > other.fNumBitVectors)
         return other.Overlap(*this);
 
-    int i;
-    for( i = 0; i < fNumBitVectors; i++ )
-    {
-        if( fBitVectors[i] & other.fBitVectors[i] )
+    for (uint32_t i = 0; i < fNumBitVectors; i++ ){
+        if (fBitVectors[i] & other.fBitVectors[i])
             return true;
     }
     return false;
 }
 
 inline hsBitVector& hsBitVector::operator=(const hsBitVector& other)
 {
-    if( this != &other )
-    {
-        if( fNumBitVectors < other.fNumBitVectors )
-        {
+    if (this != &other) {
+        if (fNumBitVectors < other.fNumBitVectors) {
             Reset();
             fNumBitVectors = other.fNumBitVectors;
             fBitVectors = new uint32_t[fNumBitVectors];
-        }
-        else
-        {
+        } else {
             Clear();
         }
 
-        int i;
-        for( i = 0; i < other.fNumBitVectors; i++ )
+        for (uint32_t i = 0; i < other.fNumBitVectors; i++)
             fBitVectors[i] = other.fBitVectors[i];
     }
     return *this;
 }
 
 inline bool hsBitVector::operator==(const hsBitVector& other) const
 {
-    if( fNumBitVectors < other.fNumBitVectors )
+    if (fNumBitVectors < other.fNumBitVectors)
         return other.operator==(*this);
-    int i;
-    for( i = 0; i < other.fNumBitVectors; i++ )
-        if( fBitVectors[i] ^ other.fBitVectors[i] )
+    uint32_t i;
+    for (i = 0; i < other.fNumBitVectors; i++)
+        if (fBitVectors[i] ^ other.fBitVectors[i])
```

Comment:
AFAICT, this is an attempted (and unnecessary) optimization of `!=`  >.>

Reply:
I was wondering what in the devil was going on here. Will fix.